src/presentation/components/Settings/AxisSetSettings.vue

```diff
@@ -264,106 +265,80 @@ export default defineComponent({
   data() {
     return {
       axisSetRepository,
-      //NOTE: initialize axis values as string because it sometimes is displayed like '1e+10'
-      displayVal: {
-        x1: '',
-        x2: '',
-        y1: '',
-        y2: '',
-      },
-      axesToDisplayValAsExponential: [] as {
-        axisSetId: number
-        axisName: 'x1' | 'x2' | 'y1' | 'y2'
-      }[],
+      axisValInputHandler,
+      errorMessage: '',
     }
   },
-  created() {
-    this.displayVal.x1 = String(this.x1Axis.value)
-    this.displayVal.x2 = String(this.x2Axis.value)
-    this.displayVal.y1 = String(this.y1Axis.value)
-    this.displayVal.y2 = String(this.y2Axis.value)
-  },
+  created() {},
   methods: {
-    updateDisplayValMultipliedByTen(axisName: 'x1' | 'x2' | 'y1' | 'y2'): void {
-      this.displayVal[axisName] = this.getDisplayValMultipliedByTen(
-        parseFloat(this.displayVal[axisName]),
-      )
-    },
-    updateDisplayValDividedByTen(axisName: 'x1' | 'x2' | 'y1' | 'y2'): void {
-      this.displayVal[axisName] = this.getDisplayValDividedByTen(
-        parseFloat(this.displayVal[axisName]),
-      )
-    },
-    getDisplayValMultipliedByTen(value: number): string {
-      if (value === 0) {
-        return '1'
+    onInputAxisVal(axisKey: AxisKey, val: string) {
+      this.axisValInputHandler.setInputValue(this.activeAxisSetId, axisKey, val)
+
+      try {
+        this.axisValInputHandler.setInputValue(
+          this.activeAxisSetId,
+          axisKey,
+          val,
+        )
+
+        this.axisValInputHandler.setConvertedAxisValToAxisSet(
+          this.activeAxisSetId,
+          axisKey,
+        )
+
+        //NOTE: update validation only when there are existing validation errors
+        if (this.errorMessage) {
+          this.errorMessage = this.axisValInputHandler.getValidationMessage(
+            this.activeAxisSetId,
+          )
+        }
+      } catch (error: unknown) {
```

Comment:
Is the type unknown? Also, is it enough to just output the error to the console?